ta/tpm_log_test/ta_entry.c

```diff
@@ -0,0 +1,177 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright (c) 2020, ARM Limited. All rights reserved.
+ */
+
+#include <ta_tpm_log.h>
+#include <string.h>
+#include <inttypes.h>
+#include <pta_system.h>
+#include <tee_internal_api.h>
+
+#define LOG_SIZE		1024
+
+static TEE_Result invoke_system_pta(uint32_t cmd_id, uint32_t param_types,
+				    TEE_Param params[TEE_NUM_PARAMS])
+{
+	const TEE_UUID system_uuid = PTA_SYSTEM_UUID;
+	TEE_TASessionHandle session = TEE_HANDLE_NULL;
+	TEE_Result res = TEE_ERROR_GENERIC;
+	uint32_t ret_origin = 0;
+
+	res = TEE_OpenTASession(&system_uuid, TEE_TIMEOUT_INFINITE,
+				0, NULL, &session, &ret_origin);
+	if (res != TEE_SUCCESS)
+		return res;
+
+	res = TEE_InvokeTACommand(session, TEE_TIMEOUT_INFINITE,
+				  cmd_id, param_types, params, &ret_origin);
+
+	TEE_CloseTASession(session);
+
+	return res;
+}
+
+/*
+ * Test case for the use of right parameters.
+ */
+static TEE_Result test_with_right_parameters(void)
+{
+	unsigned char buf[LOG_SIZE] = { 0 };
+	const uint32_t param_types = TEE_PARAM_TYPES(
+						TEE_PARAM_TYPE_MEMREF_OUTPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Param params[TEE_NUM_PARAMS] = { 0 };
+	uint32_t field = 0;
+	unsigned char *digestptr = NULL;
+
+	params[0].memref.buffer = (void *)buf;
+	params[0].memref.size = LOG_SIZE;
+
+	if (invoke_system_pta(PTA_SYSTEM_GET_TPM_EVENT_LOG,
+			      param_types, params) == TEE_SUCCESS) {
+
+		DMSG("Received event log with %i bytes",
+		    params[0].memref.size);
+		DMSG("Parsing the event log header");
+		memcpy(&field, &buf[0], sizeof(uint32_t));
+
+		DMSG("--> PCRIndex = 0x%" PRIx32, field);
```

Comment:
really need to keep these?
If so, maybe remove the `--> ` prefix and possibly remove the one at line 58, add an empty line before lines 74 and 97.